SWLOR.Game.Server/Feature/ShipModuleDefinition/BulwarkShieldGeneratorModuleDefinition.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.Collections.Generic;
+using SWLOR.Game.Server.Core.NWScript.Enum;
+using SWLOR.Game.Server.Core.NWScript.Enum.VisualEffect;
+using SWLOR.Game.Server.Entity;
+using SWLOR.Game.Server.Service;
+using SWLOR.Game.Server.Service.SkillService;
+using SWLOR.Game.Server.Service.SpaceService;
+
+namespace SWLOR.Game.Server.Feature.ShipModuleDefinition
+{
+    public class BulwarkShieldGeneratorModuleDefinition : IShipModuleListDefinition
+    {
+        private readonly ShipModuleBuilder _builder = new ShipModuleBuilder();
+
+        public Dictionary<string, ShipModuleDetail> BuildShipModules()
+        {
+            BulwarkShieldGenerator("bulwarkgen", "Bulwark Shield Generator", "Bulwark Gen", "An advanced suite of shield generators project at a wide range, protecting allied ships and reinforcing their defenses, and forcing most nearby foes to target the user. Lasts 12 seconds.", 10);
+
+            return _builder.Build();
+        }
+
+        private void BulwarkShieldGenerator(
+            string itemTag,
+            string name,
+            string shortName,
+            string description,
+            int repairAmount)
+        {
+            _builder.Create(itemTag)
+                .Name(name)
+                .ShortName(shortName)
+                .Type(ShipModuleType.RepairFieldGenerator)
+                .Texture("iit_ess_075")
+                .Description(description)
+                .PowerType(ShipModulePowerType.High)
+                .Capacitor(25)
+                .Recast(12f)
+                .CapitalClassModule()
+                .CanTargetSelf()
+                .ActivatedAction((activator, activatorShipStatus, target, targetShipStatus, moduleBonus) =>
+                {
+                    var industrialBonus = Space.GetShipStatus(activator).Industrial;
```

Comment:
This has the potential to cause a null reference exception. Space.GetShipStatus can return a null in certain cases. I believe you need to add a null check here (and anywhere else) to bail out early if you don't have a valid ShipStatus object.

Reply:
Actually, looking at it closer you shouldn't need to call this at all. You already have activatorShipStatus available to you. Just use that and you don't need the null check.

Reply:
Adjusted this.